Fixes requested by NCO on some of the resources used in operations.

aerorahul opened the pull request:
<!-- PLEASE READ -->
<!--
Before opening a PR, please note these guidelines:

- Each PR should only address ONE topic and have an associated issue
- No hardcoded or paths to personal directories should be present
- No temporary or backup files should be committed
- Any code that was disabled by being commented out should be removed
-->

**Description**
NCO requested clarification and updates to the resources for some of the jobs run in operations.
This PR addresses those comments and concerns.
Some of the changes are still being tested and may need further refinement.

Tagging @WeiWei-NCO

This work is part of #398 

**Type of change**

- [X] Bug fix (non-breaking change)

**How Has This Been Tested?**
The only way to test this is with NCO running the jobs in this PR and validating it meets their acceptance criteria.

<!-- Please describe the tests that you ran to verify your changes and on the platforms these tests were conducted. -->
<!-- Provide instructions so we can reproduce. -->
<!-- Please also list any relevant details for your test configuration -->

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--
-->
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

aerorahul committed:
fix resource allocations for some jobs that NCO flagged were allocating too many cores

aerorahul committed:
update resources for wave init, post and prep jobs in ecf scripts

aerorahul committed:
update resources for atmos gempak in ecf scripts

aerorahul committed:
update resources for atmos gempak_meta in ecf scripts

aerorahul committed:
update resources for atmos pp wafs_gcip in ecf scripts

aerorahul committed:
update resources for atmos chgres for enkf in ecf scripts

WalterKolczynski-NOAA commented on ecf/scripts/gdas/wave/prep/jgdas_wave_prep.ecf:
```diff
@@ -4,7 +4,7 @@
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
+#PBS -l place=vscatter,select=1:mpiprocs=5:ompthreads=1:ncpus=5
 #PBS -l place=excl
```
Does this job still need exclusive use of a node due to memory?

aerorahul replied:
I don't know exactly what the memory use is in this job, hence the exclusive use.

WalterKolczynski-NOAA commented on ecf/scripts/gfs/wave/prep/jgfs_wave_prep.ecf:
```diff
@@ -4,7 +4,7 @@
 #PBS -l walltime=00:30:00
-#PBS -l place=vscatter,select=1:ncpus=128
+#PBS -l place=vscatter,select=1:mpiprocs=60:ompthreads=1:ncpus=60
 #PBS -l place=excl
```
Does this job still need to be exclusive?

aerorahul replied:
same as the gdas wave prep job.

WalterKolczynski-NOAA reviewed (changes requested):
Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.

aerorahul commented:
> Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.

I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. 
This came about from conversations w/ @WeiWei-NCO

WalterKolczynski-NOAA commented:
> > Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.
> 
> I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. This came about from conversations w/ @WeiWei-NCO

I agree. My point was just that if we are requesting exclusive mode, we are "using" the other CPUs on that node, even if we aren't requesting them explicitly.

aerorahul commented:
> > > Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.
> > 
> > 
> > I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. This came about from conversations w/ @WeiWei-NCO
> 
> I agree. My point was just that if we are requesting exclusive mode, we are "using" the other CPUs on that node, even if we aren't requesting them explicitly.

Actually, you are not "using" them.  If one tries to do `mpiexec -n number_exceeding_requested_cores`, the the job will fail.

KateFriedman-NOAA commented on ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf:
```diff
@@ -4,8 +4,7 @@
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=11:ompthreads=1:ncpus=11
```
excl or memory? Wei requested every job have one of either set. The waveinit job barely needs resources so maybe `mem=1GB`?

WeiWei-NCO replied:
700MB on wcoss1. I would suggest to start with 2GB

aerorahul replied:
Resolved in a1ca249a

KateFriedman-NOAA commented on ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf:
```diff
@@ -4,8 +4,7 @@
 #PBS -l walltime=00:20:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=8:ompthreads=1:ncpus=8
```
excl or memory? Don't know it's memory needs though.

WeiWei-NCO replied:
1GB for mem

aerorahul replied:
Resolved in a1ca249a

KateFriedman-NOAA commented on ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf:
```diff
@@ -4,8 +4,7 @@
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=11:ompthreads=1:ncpus=11
```
Similar to my comment on the gdas_wave_init script, excl or memory?

WeiWei-NCO replied:
2GB

aerorahul replied:
Resolved in a1ca249a

KateFriedman-NOAA commented on ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf:
```diff
@@ -4,8 +4,7 @@
 #PBS -l walltime=03:00:00
-#PBS -l place=vscatter,select=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=8:ompthreads=1:ncpus=8
```
excl or memory?

WeiWei-NCO replied:
1GB

aerorahul replied:
Resolved in a1ca249a

KateFriedman-NOAA reviewed (commented):
Changes look good, I'll need to incorporate into rocoto workflow and test. Left questions on the waveinit and wavepostsbs resources changes: need excl or memory setting per Wei's request.

aerorahul committed:
update resources for wave jobs to include memory in ecf scripts

aerorahul committed:
update resources for more jobs to include memory in ecf scripts.  wave init jobs need modules for Intel loaded

aerorahul committed:
update resources for jgdas_atmos_tropcy_qc_reloc.ecf.  Remove developer overwrite section

aerorahul committed:
update resources for jgfs_atmos_tropcy_qc_reloc.ecf.  Remove developer overwrite section

aerorahul committed:
update resources for more jobs in ecf scripts from NCO

aerorahul committed:
update resources for more jobs in ecf scripts from NCO

aerorahul commented:
@NOAA-EMC/global-workflow-cms 
I have added a few more updates since the last review following changes from NCO.

KateFriedman-NOAA committed:
Update esfc and analysis job resources

- update the gdasesfc, gdasanal, and gfsanal job resources in ecf
scripts, based on optimization work between NCO/EMC

Refs: #399

KateFriedman-NOAA committed:
Update gdasechgres job resources

- based on optimization work between NCO/EMC, update the echgres job
resources to be more appropriate for the larger node machine

Refs: #399

aerorahul committed:
bringing in changes from @WeiWei-NCO after his testing

aerorahul commented:
@KateFriedman-NOAA @WeiWei-NCO 
I think this is current as of this evening of testing and can be merged.
There may be a few more later, but I think we are on solid ground now.

KateFriedman-NOAA committed:
Update analysis job walltime to 50mins

- adjust the analysis job walltimes to be 50mins for both the gdas and
gfs suite jobs

Refs: #399

KateFriedman-NOAA reviewed (approved):
Changes look good and conform to expected optimization outcomes between NCO/EMC.
